### PR TITLE
Push application source with SF leads

### PR DESCRIPTION
**Pull request by jpslav**
Lets admins set an "application source" for each oauth app.  When users sign up from one of those apps, this new source field is sent along with the lead to Salesforce.  

Will require us to set Tutor's source value to "Tutor Signup" after deployment

### CODE REVIEW

File: config/environments/development.rb [hunk 1]
```diff
@@ -33,6 +33,12 @@
   # Raises helpful error messages.
   config.assets.raise_runtime_errors = true
 
+  # Tell Action Mailer not to deliver emails to the real world.
+  # The :test delivery method accumulates sent emails in the
+  # ActionMailer::Base.deliveries array.  If SMTP used instead
+  # but without good credentials, results in 30 second pauses.
+  config.action_mailer.delivery_method = :test
```
Comment (Dantemss) [inline config/environments/development.rb]: Not sure if the test mailer is usually used in the dev environment?

File: db/migrate/20171226134252_add_lead_source_to_oauth_applications.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class AddLeadSourceToOauthApplications < ActiveRecord::Migration
+  def change
+    add_column :oauth_applications, :lead_application_source, :string,
+                                    null: false, default: ''
```
Comment (Dantemss) [inline db/migrate/20171226134252_add_lead_source_to_oauth_applications.rb]: Not a huge fan of the blank string as a default. I'd maybe leave the column as nullable, if some apps have no lead_application_source. But I guess if Salesforce expects a blank string that may be ok.

File: app/routines/push_salesforce_lead.rb [hunk 3]
```diff
@@ -33,7 +34,8 @@ def exec(user:, email: nil, role:, phone_number:, school:, num_students:, using_
       website: url,
       adoption_status: using_openstax,
       num_students: num_students.to_i,
-      os_accounts_id: user.id
+      os_accounts_id: user.id,
+      application_source: source_application.try(:lead_application_source) || ''
```
Comment (Dantemss) [inline app/routines/push_salesforce_lead.rb]: Is a blank string what salesforce expects when we don't know the source?

Comment (Dantemss) [reply app/routines/push_salesforce_lead.rb]: Also, does SF accept any string or just some specific strings?

Comment (jpslav) [reply app/routines/push_salesforce_lead.rb]: specific strings defined by SF admin (denver et al)

Comment (jpslav) [reply app/routines/push_salesforce_lead.rb]: blank string works when we don't know the string

File: config/environments/development.rb (hunk 1 above)
Comment (jpslav) [reply config/environments/development.rb]: without this we get 30 second hang times when there's an exception in dev mode (I fixed this in Tutor some time ago and just copied the fix over here).  If someone actually needs to test mailing they can change this to `:smtp` and provide some credentials.

**Pull request merged**